src/lib/scene/socketgraphicsitem.ts

```diff
@@ -202,22 +202,25 @@ export class SocketGraphicsItem extends GraphicsItem {
 				mouseY
 			);
 
+			// check if a cyclical connection is about to be added,
 			// this can only happen when forming a new connection so no need to check
 			// for other conditions (existing connection being removed, etc...)
-			// this means this socket type should be SocketType.Out and the other
-			// should be SocketType.In
+			//
+			// this means one socket has to be SocketType.Out and the other should be SocketType.In
 			if (
 				closeSock &&
-				this.socketType == SocketType.Out &&
-				closeSock.socketType == SocketType.In
+				(
```

Comment:
I'd extract this to a variable or two.

Reply:
Agreed. I'll shuffle it around to something more readable.